src/daemon/statusnotifier.vala

```diff
@@ -0,0 +1,229 @@
+/*
+ * This file is part of budgie-desktop
+ *
+ * Copyright Budgie Desktop Developers
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+namespace Budgie.StatusNotifier {
+	public const string WATCHER_FREEDESKTOP_DBUS_NAME = "org.freedesktop.StatusNotifierWatcher";
+	public const string WATCHER_KDE_DBUS_NAME = "org.kde.StatusNotifierWatcher";
+
+	public const string WATCHER_FREEDESKTOP_DBUS_OBJECT_PATH = "/org/freedesktop/StatusNotifierWatcher";
+	public const string WATCHER_KDE_DBUS_OBJECT_PATH = "/org/kde/StatusNotifierWatcher";
+	public const string WATCHER_BASIC_DBUS_OBJECT_PATH = "/StatusNotifierWatcher";
+
+	public struct DBusServiceInfo {
+		public string name;
+		public string object_path;
+		public string sender;
+		public string owner;
+	}
+
+	[DBus (name="org.freedesktop.StatusNotifierWatcher")]
+	public class FreedesktopWatcher : Object {
+		public string[] registered_status_notifier_items {
+			owned get {
+				string[] ret = new string[registered_services.size()];
+				for (int i = 0; i < registered_services.size(); i++) {
+					ret[i] = registered_services.get_values().nth_data(i).name;
+				}
```

Comment:
Couldn't we do something like the following?

```suggestion
				foreach (DBusServiceInfo val in registered_services.get_values()) {
				    ret += val.name;
				}
```

You also don't need to specify the array size, it can be dynamically resized by doing:

```
string[] ret = {};
```

Reply:
Technically yes - I specified the size to save cycles on resizing the array unnecessarily, but it probably doesn't matter that much.